### PR TITLE
[Feature][JsonSplit-api] api of project

### PR DESCRIPTION
**Pull request by brave-lee**
api module #5498 
sub task #5521

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 1]
```diff
@@ -268,28 +264,28 @@ public boolean hasProjectAndPerm(User loginUser, Project project, Map<String, Ob
...
 
-        Project project = projectMapper.selectById(projectId);
+        Project project = projectMapper.queryByCode(projectCode);
         boolean hasProjectAndPerm = hasProjectAndPerm(loginUser, project, result);
         if (!hasProjectAndPerm) {
             return result;
         }
         Project tempProject = projectMapper.queryByName(projectName);
-        if (tempProject != null && tempProject.getId() != projectId) {
+        if (tempProject != null && tempProject.getCode().equals(projectCode)) {
```
Comment (wen-hemin) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: not equals

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
         select
         <include refid="baseSql"/>
         from t_ds_project
-        where code = #{code}
+        where code = #{projectCode}
```
Comment (wen-hemin) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml:32]: The change of variable name is not seen in mapper class, 
Are you sure it doesn't need to be modified.

**Review by wen-hemin — CHANGES_REQUESTED** [review]
Thx, please check.

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml (hunk 2 above)
Comment (brave-lee) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml:32]: Project queryByCode(@Param("projectCode") Long projectCode);

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: yes, there should be deleted

**Review by wen-hemin — APPROVED** [review]
+1

**Pull request merged**